presto-main/src/main/java/com/facebook/presto/sql/planner/plan/SemiJoinNode.java

```diff
@@ -38,6 +38,7 @@
     private final Optional<Symbol> sourceHashSymbol;
     private final Optional<Symbol> filteringSourceHashSymbol;
     private final Optional<DistributionType> distributionType;
+    private final boolean legacySemiJoin;
```

Comment:
I don't think legacy semi join is the best name of the property. I think is should rather be `predicatePushdownEnabled`, `ternaryLogic` or `semiJoinOutputNullDistinguishable`. I imagine this option should be set (in the future) during `SemiJoinNode` creation based on analysis if `null` and `false` value distinction is required.